extensions/podman/src/podman-install.ts

```diff
@@ -0,0 +1,202 @@
+/**********************************************************************
+ * Copyright (C) 2022 Red Hat, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ ***********************************************************************/
+
+import { downloadFile, fetchJson, isMac, isWindows } from './util';
+import * as fs from 'node:fs';
+import * as os from 'node:os';
+import * as path from 'node:path';
+import { spawn } from 'node:child_process';
+import * as extensionApi from '@tmpwip/extension-api';
+import { rcompare } from 'semver';
+
+import * as podmanTool from './podman.json';
+
+interface PodmanAsset {
+  name: string;
+  browser_download_url: string;
+}
+
+interface PodmanRelease {
+  body: string;
+  name: string;
+  assets: PodmanAsset[];
+}
+
+export async function installBundledPodman(): Promise<boolean> {
```

Comment:
it's quite hard to know what is done with all this if everywhere

I think we should have a common interface with different implementation (Windows/Mac and Linux)

and a `supportsInstallation` like method to not mix the 'installation' and if it's supported or not